skia-safe/src/modules/image_asset.rs

```diff
@@ -0,0 +1,116 @@
+use crate::Data;
+use crate::{prelude::*, Image, Matrix, SamplingOptions};
+use skia_bindings::{
+    self as sb, skresources_ImageAsset, skresources_ImageAsset_FrameData,
+    skresources_ImageAsset_SizeFit, C_ImageFrameData_Make, C_RustImageAsset_New, RustImageAsset,
+    RustImageAsset_Param, SkRefCnt, SkRefCntBase, TraitObject,
+};
+
+pub use sb::skresources_ImageDecodeStrategy as ImageDecodeStrategy;
+
+pub type ImageAsset = RCHandle<skresources_ImageAsset>;
+require_base_type!(skresources_ImageAsset, SkRefCnt);
+
+impl NativeRefCountedBase for skresources_ImageAsset {
+    type Base = SkRefCntBase;
+}
+
+impl ImageAsset {
+    pub fn is_multi_frame(&self) -> bool {
+        unsafe { sb::C_ImageAsset_isMultiFrame(self.native_mut_force()) }
+    }
+
+    pub fn from_data(
+        data: impl Into<Data>,
+        decode_strategy: impl Into<Option<ImageDecodeStrategy>>,
+    ) -> Option<Self> {
+        let decode_strategy = decode_strategy
+            .into()
+            .unwrap_or(ImageDecodeStrategy::LazyDecode);
+
+        ImageAsset::from_ptr(unsafe {
+            sb::C_MultiFrameImageAsset_Make(data.into().into_ptr(), decode_strategy)
+        })
+    }
+
+    pub fn from_custom_image_asset(
+        custom_image_asset: impl Into<NativeImageAsset>,
+    ) -> Option<Self> {
+        ImageAsset::from_ptr(unsafe {
+            std::mem::transmute::<_, *mut skresources_ImageAsset>(custom_image_asset.into())
+        })
+    }
+}
+
+pub type ImageSizeFit = skresources_ImageAsset_SizeFit;
+pub type ImageFrameData = skresources_ImageAsset_FrameData;
+
+pub fn create_image_frame_data(
```

Comment:
I thought it would be possible to create the FrameData struct as a rust struct but it uses skia shared pointer and it looks like there is no way to get the shared pointer from the `RCHandle` without getting into the skia world. Correct me if I wrong

Reply:
Yes, we could provide a generic function for that, but this would in turn mean that we directly handle sp typed smartpointers in Rust, which we don't. There might be a way to create a compatible struct in Rust and placing a RCHandle at the same field position, but this feels scary. Speaking of that, FrameData should be wrapped and made opaque to the Rust world (with functions wrapping access to the fields), otherwise clients could mess around with the smartpointer.